packages/opentelemetry-core/src/propagation/composite.ts

```diff
@@ -69,9 +69,11 @@ export class CompositePropagator implements TextMapPropagator {
       try {
         propagator.inject(context, carrier, setter);
       } catch (err) {
-        diag.warn(
-          `Failed to inject with ${propagator.constructor.name}. Err: ${err.message}`
-        );
+        if (err instanceof Error) {
+          diag.warn(
+            `Failed to inject with ${propagator.constructor.name}. Err: ${err.message}`
```

Comment:
this might be swallowed silently I would rather do
```suggestion
            `Failed to inject with ${propagator.constructor.name}. Err: ${err?.message}`
```
instead of checking for instance. Would rather see such message
`Failed to inject with SomeContructorName. Err: `
then not seeing message at all

Reply:
This fix doesn't work because `unknown` isn't treated like `any` if you use `?.`. You would need a cast

Reply:
What about moving this into some helper function to have the checks/casts only at one location?

Reply:
Maybe now is a good time to introduce some structured logging to the diag logger? There could be advantages to being able to call ```api.diag.error({ message: `Failed to inject with ${propagator.constructor.name}`, error })```.